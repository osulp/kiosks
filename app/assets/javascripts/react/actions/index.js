export * from './kioskActions';
export * from './touchActions';
