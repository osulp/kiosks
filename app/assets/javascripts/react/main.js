import React from "react";
import ReactDOM from "react-dom";
import Root from "./components/Root";

require("../../stylesheets/application.scss");
const root_dom_element = document.getElementById('application_root');

const setSlides = (slides) => {
  if(!slides) {
    return [];
  }
  let json = JSON.parse(slides);
  return json.slides ? json.slides : json;
};

const setRestartKiosk = (restart_kiosk) => {
  if(!restart_kiosk) {
    return [];
  }
  let json = JSON.parse(restart_kiosk);
  return json.restart_kiosk ? json.restart_kiosk : json;
};


const setMaps = (maps) => {
  if(!maps) {
    return [];
  }
  let json = JSON.parse(maps);
  return json.maps ? json.maps : json;
};

if (root_dom_element) {
  let slides = setSlides(root_dom_element.getAttribute('data-slides'));
  let restart_kiosk = setRestartKiosk(root_dom_element.getAttribute('data-restart-kiosk'));
  let maps = setMaps(root_dom_element.getAttribute('data-maps'));
  let map_default_floor_number = root_dom_element.getAttribute('data-kiosk-map-default-floor-number');
  let kiosk_type = root_dom_element.getAttribute('data-kiosk-type');
  let kiosk_url = root_dom_element.getAttribute('data-kiosk-url');
  let ga = window.ga;

  // render the root container with properties
  ReactDOM.render(
    <Root slides={slides}
          restart_kiosk={restart_kiosk}
          maps={maps}
          kiosk_map_default_floor_number={map_default_floor_number}
          kiosk_type={kiosk_type}
          kiosk_url={kiosk_url}
          google_analytics={ga}
    />,
    root_dom_element
  );
}
