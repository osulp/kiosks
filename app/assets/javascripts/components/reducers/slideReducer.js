import { SET_SLIDES, REFRESH_SLIDES } from '../actions/slideActions';

const slideReducer = (state = {}, action) => {
   switch (action.type) {
    case REFRESH_SLIDES:
      return Object.assign({}, action.slides);
    case SET_SLIDES:
      return action.slides;
    default:
      return state;
  }
}

export default slideReducer;