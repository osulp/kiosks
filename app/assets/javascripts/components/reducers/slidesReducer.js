import { SET_SLIDES, REFRESH_SLIDES } from '../actions/SlideActions';

const slidesReducer = (state = {}, action) => {
   switch (action.type) {
    case REFRESH_SLIDES:
      return Object.assign({}, action.slides);
    case SET_SLIDES:
      return action.slides;
    default:
      return state;
  }
}

export default slidesReducer;