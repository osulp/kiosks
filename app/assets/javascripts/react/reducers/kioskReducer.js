import {SCROLL_TO_SLIDE, SET_SLIDES, SET_KIOSK, ADD_ERROR, SET_TITLE, SET_GOOGLE_ANALYTICS} from '../actions/kioskActions';

export const initial_state = {
  type: "touch",
  url: "kiosk/touch",
  api: {
    hours: "/api/v1/hours",
    classroom_schedule: "/api/v1/classrooms/date/{date}",
    classrooms: "/api/v1/classrooms/rooms",
    available_rooms: "/api/v1/rooms/available/{date}"
  },
  title: "Donor Impact",
  slides: [],
  starting_slide_index: 0,
  errors: [],
  google_analytics: undefined
};

const kioskReducer = (state = initial_state, action) => {
  switch (action.type) {
    case ADD_ERROR:
      return Object.assign({}, state, { errors: [...state.errors, action.error] });
    case SET_KIOSK:
      return Object.assign({}, state, { type: action.kiosk.type, url: action.kiosk.url });
    case SET_SLIDES:
      return Object.assign({}, state, { slides: action.kiosk.slides });
    case SCROLL_TO_SLIDE:
      return Object.assign({}, state, { starting_slide_index: action.index });
    case SET_TITLE:
      return Object.assign({}, state, { title: action.title });
    case SET_GOOGLE_ANALYTICS:
      return Object.assign({}, state, { google_analytics: action.google_analytics });
    default:
      return state;
  }
};

export default kioskReducer;
