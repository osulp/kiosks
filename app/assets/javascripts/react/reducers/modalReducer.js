import {SET_MODAL_ROOT_COMPONENT, SET_MODAL_VISIBILITY} from '../actions/modalActions';

export const initial_state = {
  visible: false,
  root_component: null
};

const modalReducer = (state = initial_state, action) => {
  switch (action.type) {
    case SET_MODAL_VISIBILITY:
      return Object.assign({}, state, { visible: action.visible });
    case SET_MODAL_ROOT_COMPONENT:
      return Object.assign({}, state, { root_component: action.root_component });
    default:
      return state;
  }
};

export default modalReducer;