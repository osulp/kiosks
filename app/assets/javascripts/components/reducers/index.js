import { combineReducers } from 'redux';
import slidesReducer from './slidesReducer';

const rootReducer = combineReducers({
  slides: slidesReducer
});

export default rootReducer;