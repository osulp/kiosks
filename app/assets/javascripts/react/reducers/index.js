import { combineReducers } from 'redux';
import kioskReducer from './kioskReducer';
import modalReducer from './modalReducer';
import touchReducer from './touchReducer';

export const REDUCERS_SHAPE = {
  kiosk: kioskReducer,
  modal: modalReducer,
  touch: touchReducer
};

export default combineReducers(REDUCERS_SHAPE);