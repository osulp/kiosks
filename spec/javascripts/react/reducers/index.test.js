import {REDUCERS_SHAPE} from '../../../../app/assets/javascripts/react/reducers';
import * as combinedReducers from '../../../../app/assets/javascripts/react/reducers';

describe('Reducer::Root', () => {
  describe('Redux combinedReducers', () => {
    it('matches the snapshot', () => {
      expect(combinedReducers).toMatchSnapshot();
    });
  });
  describe('Reducers state shape', () => {
    it('matches the snapshot', () => {
      expect(REDUCERS_SHAPE).toMatchSnapshot();
    });
  });
});
