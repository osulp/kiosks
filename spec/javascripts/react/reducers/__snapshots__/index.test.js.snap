exports[`Reducer::Root Reducers state shape matches the snapshot 1`] = `
Object {
  "kiosk": [Function],
  "modal": [Function],
  "touch": [Function],
}
`;

exports[`Reducer::Root Redux combinedReducers matches the snapshot 1`] = `
Object {
  "REDUCERS_SHAPE": Object {
    "kiosk": [Function],
    "modal": [Function],
    "touch": [Function],
  },
  "default": [Function],
}
`;
